app/repositories/OptOutSessionDataRepository.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import cats.data.OptionT
+import models.incomeSourceDetails.{TaxYear, UIJourneySessionData}
+import models.itsaStatus.ITSAStatus
+import models.itsaStatus.ITSAStatus.ITSAStatus
+import models.optout.OptOutSessionData
+import play.api.libs.json.{Json, OFormat}
+import repositories.OptOutContextData.{statusToString, stringToStatus}
+import services.optout.OptOutProposition
+import services.optout.OptOutProposition.createOptOutProposition
+import uk.gov.hmrc.http.HeaderCarrier
+import utils.OptOutJourney
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class OptOutSessionDataRepository @Inject()(val repository: UIJourneySessionDataRepository) {
```

Comment:
* this seems to me a service rather than repository;
* we are moving away from UIJourneySessionDataRepository in favour of SensitiveUIJourneySessionDataRepository;

Reply:
It feels more like a repository to me as its role is to store and retrieve journey data. It's a wrapping layer around the generic journey repository that provides Opt Out specific functionality, and reduces the size of the large `OptOutService` class.

We have no sensitive data in this journey, so it seems unnecessary to migrate to storing this as sensitive data.

Reply:
@ian-hmrc - as I understand UIJourneySessionDataRepository will be decommission, you really need to have your own repository in this case

Reply:
OK. Thanks @avelx. I'll raise a tech-debt ticket to migrate to the Sensitive repo.

Reply:
I've created [MISUV-8248](https://jira.tools.tax.service.gov.uk/browse/MISUV-8248) for this.